### PR TITLE
Zer 85 Move report generation to worker

### PR DESCRIPTION
**Pull request by AnastasiiaBilynska**
dev
## JIRA

* [ZER-85](https://jira.softserve.academy/browse/ZER-85)

## Code reviewers

- [ ] @github_username

### Second Level Review

- [ ] @github_username

## Summary of issue

ToDo

## Summary of change

ToDo

## Testing approach

ToDo

## CHECK LIST
- [ ]  СI passed
- [ ]  Сode coverage >=95%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

File: config/environments/development.rb [hunk 1]
```diff
@@ -73,10 +73,11 @@
...
   # Use an evented file watcher to asynchronously detect changes in source code,
   # routes, locales, etc. This feature depends on the listen gem.
-  config.file_watcher = ActiveSupport::EventedFileUpdateChecker
+  #config.file_watcher = ActiveSupport::EventedFileUpdateChecker
 
   # Uncomment if you wish to allow Action Cable access from any origin.
   # config.action_cable.disable_request_forgery_protection = true
 
   config.action_mailer.default_url_options = { host: 'localhost', port: 3000 }
+  config.active_job.queue_adapter = :sidekiq
```
Comment (olegsobchuk) [inline config/environments/development.rb:82]: don't duplicate this string via all environments. Just add it to `application.rb`

Comment (AnastasiiaBilynska) [reply config/environments/development.rb]: I deleted it.

File: spec/jobs/user_report_job_spec.rb [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+      }.to have_enqueued_job
+    end
+    it "uploads a report" do
+      ActiveJob::Base.queue_adapter = :test
+      expect {
+        UserReportJob.set(wait_until: Date.tomorrow.noon, queue: "low").perform_later('report')
+      }.to have_enqueued_job.with('report').on_queue("low").at(Date.tomorrow.noon)
+    end
+  end
+end
```
Comment (olegsobchuk) [inline spec/jobs/user_report_job_spec.rb:20]: add empty line in the end of file

File: spec/jobs/user_report_job_spec.rb [hunk 3]
```diff
@@ -0,0 +1,20 @@
+require 'rails_helper'
+require 'sidekiq/testing'
+Sidekiq::Testing.fake!
+
+RSpec.describe UserReportJob, type: :job do
+  describe "#perform_later" do
+    it "uploads a report" do
+      ActiveJob::Base.queue_adapter = :test
```
Comment (olegsobchuk) [inline spec/jobs/user_report_job_spec.rb:8]: this string should be in spec_helper

File: config/routes.rb [hunk 4]
```diff
@@ -1,6 +1,9 @@
 # frozen_string_literal: true
 
+require 'sidekiq/web'
+
 Rails.application.routes.draw do
+  mount Sidekiq::Web => '/sidekiq'
```
Comment (olegsobchuk) [inline config/routes.rb:6]: pls, add basic auth to this page. Because now it's accessible for everyone

**Review by olegsobchuk — COMMENTED** [review]
I don'r see settings for Sidekiq server

File: config/routes.rb [hunk 5]
```diff
@@ -1,6 +1,12 @@
 # frozen_string_literal: true
 
+require 'sidekiq/web'
+
 Rails.application.routes.draw do
+  authenticate :user do
```
Comment (olegsobchuk) [inline config/routes.rb:6]: suppose `user` shouldn't see this page, but admin

File: spec/jobs/user_report_job_spec.rb [hunk 6]
```diff
@@ -0,0 +1,18 @@
+require 'rails_helper'
+require 'sidekiq/testing'
+Sidekiq::Testing.fake!
+
+RSpec.describe UserReportJob, type: :job do
+  describe "#perform_later" do
+    it "uploads a report" do 
+      expect {
+        UserReportJob.perform_later('report')
```
Comment (olegsobchuk) [inline spec/jobs/user_report_job_spec.rb:9]: move `UserReportJob.perform_later('report')` to `let` or `subject`

File: spec/jobs/user_report_job_spec.rb [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+      }.to have_enqueued_job
+    end
+    it "uploads a report" do
+      expect {
+        UserReportJob.set(wait_until: Date.tomorrow.noon, queue: "low").perform_later('report')
```
Comment (olegsobchuk) [inline spec/jobs/user_report_job_spec.rb:14]: move `UserReportJob.set(wait_until: Date.tomorrow.noon, queue: "low").perform_later('report')` to `let` or `subject`

File: config/initializers/sidekiq.rb [hunk 8]
```diff
@@ -0,0 +1,9 @@
+# frozen_string_literal: true
+
+Sidekiq.configure_client do |config|
+  config.redis = { size: 1 }
+end
+
+Sidekiq.configure_server do |config|
+  config.redis = { size: 7 }
+end
```
Comment (olegsobchuk) [inline config/initializers/sidekiq.rb:9]: it's correct, BUT
1. please, add `url` and `namespace` parameter to be more obvious
2. `url` should use `ENV` variable

File: config/initializers/devise_async.rb [hunk 9]
```diff
@@ -0,0 +1,3 @@
+Devise::Async.setup do |config|
```
Comment (olegsobchuk) [inline config/initializers/devise_async.rb:1]: can we use only `devise.rb` file for this?

**Review by olegsobchuk — COMMENTED** [review]
Check

File: config/initializers/devise_async.rb (hunk 9 above)
Comment (AnastasiiaBilynska) [reply config/initializers/devise_async.rb]: Yes, I corrected this

File: config/routes.rb (hunk 5 above)
Comment (AnastasiiaBilynska) [reply config/routes.rb]: I couldn't test it as an admin. I corrected it.

**Review by olegsobchuk — APPROVED** [review]

**Review by Leeroy6821 — APPROVED** [review]
.

**Pull request merged**